doc/source/whatsnew/v0.24.0.txt

```diff
@@ -288,9 +288,7 @@ Bug Fixes
 Categorical
 ^^^^^^^^^^^
 
--
--
--
+- Bug in :meth:`Categorical.from_codes` where ``NaN`` values in `codes` were silently converted to ``0`` (:issue:`21767`)
```

Comment:
We're also changing the behavior of `.from_codes([1.1, 2.0])`. Make a note of that here.

Also note that both of these cases now raise a `ValueError`.